### PR TITLE
Fix build_amd.py for hip-clang

### PR DESCRIPTION
**Pull request by yxsamliu**

### CODE REVIEW

File: tools/amd_build/build_amd.py [hunk 1]
```diff
@@ -141,10 +141,17 @@
...
+        return False
+
 hipify_python.hipify(
     project_directory=proj_dir,
     output_directory=out_dir,
     includes=includes,
     ignores=ignores,
     out_of_place_only=args.out_of_place_only,
-    hip_clang_launch=args.hip_clang_launch)
+    hip_clang_launch=args.hip_clang_launch or is_hip_clang())
```
Comment (iotamudelta) [inline tools/amd_build/build_amd.py]: Is there a need to allow for an override still or should it just be `hip_clang_launch=is_hip_clang()`?

Comment (yxsamliu) [reply tools/amd_build/build_amd.py]: no. except to keep some existing script happy. would you like me to clean up the option?

Comment (iotamudelta) [reply tools/amd_build/build_amd.py]: if you could clean it up, that'd be nice. assuming of course that you can guarantee that the `is_hip_launch()` function will be reliably detecting `hip-clang` moving forward.

Comment (yxsamliu) [reply tools/amd_build/build_amd.py]: I updated the PR to remove the old option. I am pretty sure this is the stable way to identify hip-clang.

File: tools/amd_build/build_amd.py [hunk 2]
```diff
@@ -141,10 +135,17 @@
                     f.flush()
                     os.fsync(f)
 
+# Check if the compiler is hip-clang.
+def is_hip_clang():
+    try:
+        return 'HIP_COMPILER=clang' in open('/opt/rocm/hip/lib/.hipInfo').read()
```
Comment (mhbliao) [inline tools/amd_build/build_amd.py]: We should avoid checking absolute path. It may conflicts with custom environment from a developer.

Comment (mhbliao) [reply tools/amd_build/build_amd.py]: even though that works for the current CI setting, it may conflict with other developers environment

Comment (yxsamliu) [reply tools/amd_build/build_amd.py]: updated to use HIP_PATH environment variable.

**Comment by yxsamliu** [comment]
can we merge this PR? hip-clang framework-ci is blocked on this change. the failure seems not related to this change.

**Comment by iotamudelta** [comment]
@pytorchbot retest this please

For some reason this PR keeps failing.

**Pull request merged**